### DISCUSSION TITLE
Issues with RDP

### DISCUSSION THREAD

**Discussion by teknixstuff** [Maplespe/DWMBlurGlass#160] [Q&A]
When multiple sessions are open via RDP, DWMBlurGlass seems to fail injecting into any DWM processes.
The bug can probably also happen with multiple users logged in locally, but this is untested.

**Comment by teknixstuff** [comment]
After additional testing, it seems that injecting the DLL into the DWM process for an RDP session crashes the DWM process, while injecting into the console session's DWM process causes no issues and works as expected.

**Comment by teknixstuff** [comment]
After more testing it seems that injecting into any process crashes it.

**Comment by Maplespe** [comment]
I don't think RDP should use the Blur effect, the default behavior of the system is to apply a Basic theme to increase fluidity.

**Comment by teknixstuff** [comment]
I've finally figured out what happens: DWMBlurGlass seems to only inject into the first DWM process. It would be better to inject into all DWM processes (or at least add an option to do so)

**Comment by teknixstuff** [comment]
> I don't think RDP should use the Blur effect, the default behavior of the system is to apply a Basic theme to increase fluidity.

Yes, back in Vista/7 that was the case. However, since 8, there is no basic theme to be applied. Most W7 themes for W10 (or at least the one I use) do not look good without glass, and automatically switching themes is difficult to do.

**Comment by Maplespe** [comment]
In fact in win7 switching the basic theme was due to disabling compositing and aero effects.
And since win8, the window itself is not transparent so there is no need to switch the basic theme.

**Comment by teknixstuff** [comment]
RDP isn't the only case where this issue happens. The issue will also happen if multiple users are logged in locally (eg fast use switching).
